src/getPageObject.ts

```diff
@@ -31,10 +34,25 @@ export default async function getPageObject({
     if (!page.client.default) {
       throw new InternalError('No default export found for given route');
     }
-    const appFile = getAppFile({ options });
     return { page, appFile, ...routeInfo };
   }
-  throw new InternalError('No matching page found for given route');
+
+  // 404
+  const { route } = options;
+  const { pathname, search } = parseRoute({ route });
+  const query = parseQueryString({ queryString: search });
+
+  return {
+    appFile,
+    pagePath: pathname,
+    params: {},
+    paramsNumber: Object.keys(query).length,
+    query,
+    resolvedUrl:
+      pathname +
+      stringifyQueryString({ object: query, leadingQuestionMark: true }),
+    route,
+  };
 }
```

Comment:
Let's refactor `getPageObject.ts` at the end if this development cycle. I think that with the right abstraction we might avoid some duplication. Good so far.

Reply:
Lets keep refactoring our of this PR to keep the diff smaller and do a separate PR after this is merged and before we merge into `master`. WDYT?

Reply:
Yo! :)